### ISSUE TITLE
Problem in execute

### ISSUE THREAD

**Issue by jessiffmm** [JdeRobot/dl-DetectionSuite#90]
Hi,

I'm trying to execute with:
./DatasetEvaluationApp -c myappConfig.yml
But I get:
./DatasetEvaluationApp: error while loading shared libraries: libopencv_core3.so.3.3: cannot open shared object file: No such file or directory

I know that libopencv_core3.so.3.3 is in /opt/ros/kinetic/lib/x86_64-linux-gnu but I don't understand why I get this error.

Thanks!! Regards!

**Comment by vinay0410** [comment]
Hi @jessiffmm 
Have you installed opencv separartely or are you using ros's OpenCV.
This is probably an issue caused by ROS's opencv.

I have updated the pull #88 . This will shift support completely to OpenCV 3.4 and above, and therefore won't cause such conflicts in the future.

Also, we have appimages available at releases [here](https://github.com/JdeRobot/dl-DetectionSuite/releases). You can directly run Dataset Evaluation App using them without having to run anything.

**Comment by jessiffmm** [comment]
Hi @vinay0410 

I get the same error after doing git pull. I installed OpenCV 3.4 separately. I don't understand why It looks for  libopencv_core3.so.3.3 if It's running with OpenCV 3.4.

How Can I execute appimages? I had tried running Dataset Evaluation App with DetectionSuite-x86_64.AppImage and DetectionSuite_with_ROS_and_ICE-x86_64.AppImage, but I couldn't .

**Comment by vinay0410** [comment]
To execute appImage, download and git it executable permissions.

```chmod a+x DetectionSuite-x86_64.AppImage```

Then just run it.
```./DetectionSuite-x86_64.AppImage -c myappConfig.yml```

And, for OpenCV error, I have submitted a pull, let it merge and then you can try again.
Also, to rebuild please delete the previous build directory and start afresh.

**Comment by vinay0410** [comment]
Hi @jessiffmm ,
The pull request has been merged, I couldn't reproduce any error after this merge, but if you do , just let me know.

**Comment by jessiffmm** [comment]
Hi @vinay0410 ,

Thanks!
Finally I have able to install detection suite after reinstalling ubuntu again.
Where Can I get a manual or something similar about detection suite?

**Comment by vinay0410** [comment]
Usage Documentation is available in this project’s Wiki.
You can start with the [beginner’s guide ](https://github.com/JdeRobot/dl-DetectionSuite/wiki/Beginner's-Tutorial-to-DetectionSuite-Part-1) which has some detailed instructions about running DetectionSuite.

**Comment by jessiffmm** [comment]
Okk Thanks!!

I tried to upload a video but I get  a error. Do I have to use some video format?

**Comment by vinay0410** [comment]
Depends, on  the error, if it is a codec error, then video format might not be supported.
But, since it uses OpenCV to read from videos, that is highly unlikely.

Make sure, to select all the parameters.
And , please mention the exact error you are facing.

**Comment by jessiffmm** [comment]
Hi,

I get:
Unable to stop the stream: Función ioctl no apropiada para el dispositivo
WARNING: Logging before InitGoogleLogging() is written to STDERR
W1017 23:16:34.676131  7739 Deployer.cpp:26] Error Creating Generic Live Reader
Error Message: Couldn't open Video file!

I try to upload a MPG video.

**Comment by jessiffmm** [comment]
Also, When I use tensorflow I get the next error:
    raise ImportError('Please upgrade your tensorflow installation to v1.4.* or later!')
ImportError: Please upgrade your tensorflow installation to v1.4.* or later!
Cannot find function "tensorflow_detect"
Detection Graph Loaded
Evaluating : 1(1/0)
/tmp/.mount_DetecthQqT5t/AppRun: línea 12:  8892 Violación de segmento  (`core' generado) LD_LIBRARY_PATH="$DIR/usr/lib" "${DIR}/usr/bin/DatasetEvaluationApp" 

I don't understand why I get this error. I execute print tf.__version__ and I get 1.11.0.

**Comment by jessiffmm** [comment]
If I use yolo and camera (deployer input type) I get:
WARNING: Logging before InitGoogleLogging() is written to STDERR
I1017 23:53:08.768757  9033 CameraReader.cpp:4] Starting Capture from device with DeviceId: 0
Config file has no sections: Invalid argument

If I use keras I get:
Traceback (most recent call last):
  File "/tmp/.mount_Detectq0r7k2/usr/lib/python_modules/keras_detect.py", line 29, in __init__
    'compute_loss': ssd_loss.compute_loss})
  File "/usr/local/lib/python2.7/dist-packages/keras/engine/saving.py", line 417, in load_model
    f = h5dict(filepath, 'r')
  File "/usr/local/lib/python2.7/dist-packages/keras/utils/io_utils.py", line 186, in __init__
    self.data = h5py.File(path, mode=mode)
  File "/home/vanejessi/.local/lib/python2.7/site-packages/h5py/_hl/files.py", line 312, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/home/vanejessi/.local/lib/python2.7/site-packages/h5py/_hl/files.py", line 142, in make_fid
    fid = h5f.open(name, flags, fapl=fapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5f.pyx", line 78, in h5py.h5f.open
IOError: Unable to open file (file signature not found)

**Comment by vinay0410** [comment]
Hi @jessiffmm ,
Thanks for reporting all those issues !

1. There might be specific Video formats from which OpenCV's Videocapture can read, and MPG may not be supported, you might be facing a different issue. Just try [this video](https://drive.google.com/open?id=17b3cG8H___yjxcoFcdTe1vPEIccvKt6D), which I have tested on my build.

2. That is because of a string comparison between 1.4.1 & 1.11.0, and 1.11.0 comes out to be smaller, so I have fixed that, and will not occur again. This #92 fixes this bug.

3. In keras you've to pass ```.h5``` models and **not** tensorflow inference graphs (.pb). Try keras 
again with a .h5 model.
Try [this one](http://jderobot.org/store/deeplearning-networks/Keras/VGG_300_300_coco.h5)

Also, just inform me when you are online and using DetectionSuite, so if any more errors occur I will get them resolved then and there.
